### PR TITLE
Reworded the "How to use Gmail" cookbook article

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | all |
| Fixed tickets | - |

In StackOverflow some people are [having issues](http://stackoverflow.com/questions/33226583/swiftmailer-doesnt-work/33240538#33240538) with the `encryption` parameter of Gmail. And others are [having issues](https://github.com/symfony/swiftmailer-bundle/issues/106) with Gmail + SwiftMailerBundle.

That's why I propose to reword this article and explain some things with more detail.

### CODE REVIEW

File: cookbook/email/gmail.rst [hunk 1]
```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
             'password'  => 'your_gmail_password',
         ));
 
-You're done!
+If you are using the Symfony Standard Edition, it's more convenient to configure
+these parameters in ``parameters.yml``:
```
Comment (WouterJ) [inline cookbook/email/gmail.rst]: "these parameters" seems wrong, as we never talked about parameters before in this article. What about: "*If you're using the Symfony Standard Edition, it's more convenient to use parameters for these options:*"

File: cookbook/email/gmail.rst [hunk 2]
```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
...
 
-.. tip::
+.. code-block:: yaml
 
-    If you are using the Symfony Standard Edition, configure the parameters in ``parameters.yml``:
+    # app/config/parameters.yml
```
Comment (WouterJ) [inline cookbook/email/gmail.rst]: I wonder if this shouldn't be put in `parameters.yml.dist` instead?

File: cookbook/email/gmail.rst [hunk 3]
```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
...
-            # ...
-            mailer_transport: gmail
-            mailer_host:      ~
-            mailer_user:      your_gmail_username
-            mailer_password:  your_gmail_password
+        # app/config/config_dev.yml
+        swiftmailer:
+            transport: gmail
+            username:  %mailer_user%
+            password:  %mailer_password%
```
Comment (WouterJ) [inline cookbook/email/gmail.rst]: double quotes are needed in these 2 lines (`"%mailer_user%", `"%mailer_password%"``

File: cookbook/email/gmail.rst (hunk 1 above)
Comment (OskarStark) [inline cookbook/email/gmail.rst]: ...to use parameters.yml for these options...?

File: cookbook/email/gmail.rst (hunk 2 above)
Comment (OskarStark) [inline cookbook/email/gmail.rst]: maybe adding a note/tip to add this in the dist file too?

File: cookbook/email/gmail.rst [hunk 4]
```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
...
+            'password'  => '%mailer_password%',
+        ));
+
+Redefining the Default Configuration Parameters
+-----------------------------------------------
 
-.. note::
+The ``gmail`` transport is simply a shortcut that uses the ``smtp`` transport
+and sets ``encryption`` to ``ssl``, ``auth_mode`` to ``login`` and ``host`` to
+``smtp.gmail.com`` to work with Gmail.
```
Comment (OskarStark) [inline cookbook/email/gmail.rst]: what about using a table here for the options?

File: cookbook/email/gmail.rst [hunk 5]
```diff
@@ -55,33 +50,73 @@ In the development configuration file, change the ``transport`` setting to
...
 
-    The ``gmail`` transport is simply a shortcut that uses the ``smtp`` transport
-    and sets ``encryption``, ``auth_mode`` and ``host`` to work with Gmail.
+If your application uses ``tls`` encryption or ``oauth`` authentication, you
+must override the default options by defining the ``encryption`` and ``auth_mode``
+parameters.
 
-.. note::
+If your Gmail account uses 2-Step-Verification, you must `generate an App password`_
+to use for your ``mailer_password`` parameter. You must also ensure that you
```
Comment (OskarStark) [inline cookbook/email/gmail.rst]:
~~~
If your Gmail account uses 2-Step-Verification, you must `generate an App password`_
to use for your ``mailer_password`` parameter.
~~~

sounds weird, what about:
~~~
If you are using 2-Step-Verification, you must `generate an App password`_ and use this as your ``mailer_password``.
~~~

**Comment by OskarStark** [comment]
Great work @javiereguiluz

**Comment by javiereguiluz** [comment]
I've changed everything I was asked for. Time for another review. Thanks.

**Comment by OskarStark** [comment]
nice :+1:

File: cookbook/email/gmail.rst (hunk 3 above)
Comment (xabbuh) [inline cookbook/email/gmail.rst]: We should use single quotes instead.

Comment (javiereguiluz) [inline cookbook/email/gmail.rst]: Thanks for the heads up. I've updated this. Hopefully this PR can now be considered finished.

File: cookbook/email/gmail.rst [hunk 6]
```diff
@@ -55,33 +50,82 @@ In the development configuration file, change the ``transport`` setting to
...
+==============  ==================
 
-.. note::
+If your application uses ``tls`` encryption or ``oauth`` authentication, you
+must override the default options by defining the ``encryption`` and ``auth_mode``
+parameters.
 
-    The ``gmail`` transport is simply a shortcut that uses the ``smtp`` transport
-    and sets ``encryption``, ``auth_mode`` and ``host`` to work with Gmail.
+If you are using 2-Step-Verification, you must `generate an App password`_ and use this as your ``mailer_password`` value.
```
Comment (xabbuh) [inline cookbook/email/gmail.rst]: looks like you forgot to wrap the sentence :)

File: cookbook/email/gmail.rst [hunk 7]
```diff
@@ -55,33 +50,82 @@ In the development configuration file, change the ``transport`` setting to
...
+If your Gmail account uses 2-Step-Verification, you must `generate an App password`_
+and use it as the value of the ``mailer_password`` parameter. You must also ensure
+that you `allow less secure apps to access your Gmail account`_.
 
-    Depending on your Gmail account settings, you may get authentication errors
-    within your app. If your Gmail account uses 2-Step-Verification, you should
-    `generate an App password`_ to use for your ``mailer_password`` parameter.
-    You should also ensure that you `allow less secure apps to access your Gmail account`_.
+See the :doc:`Swiftmailer configuration reference </reference/configuration/swiftmailer>`
+for more details.
```
Comment (xabbuh) [inline cookbook/email/gmail.rst]: I would wrap this in a `seealso` block.

**Comment by javiereguiluz** [comment]
I've implemented the last round of fixes. Please tell me that this is ready for the merge :relieved:

File: cookbook/email/gmail.rst [hunk 8]
```diff
@@ -55,33 +50,84 @@ In the development configuration file, change the ``transport`` setting to
...
+that you `allow less secure apps to access your Gmail account`_.
 
-.. note::
+.. seealso::
 
-    Depending on your Gmail account settings, you may get authentication errors
-    within your app. If your Gmail account uses 2-Step-Verification, you should
-    `generate an App password`_ to use for your ``mailer_password`` parameter.
-    You should also ensure that you `allow less secure apps to access your Gmail account`_.
+    :doc:`Swiftmailer configuration reference </reference/configuration/swiftmailer>`
```
Comment (xabbuh) [inline cookbook/email/gmail.rst]: I would have kept the sentence though:

```rst
.. seealso::

    See the :doc:`Swiftmailer configuration reference </reference/configuration/swiftmailer>`
    for more details.
```

Comment (xabbuh) [inline cookbook/email/gmail.rst]: But we can make that change when merging. The PR so far looks ready to me. :+1:

File: cookbook/email/gmail.rst [hunk 9]
```diff
@@ -55,33 +50,84 @@ In the development configuration file, change the ``transport`` setting to
             'password'  => 'your_gmail_password',
         ));
 
-You're done!
+If you are using the Symfony Standard Edition, it's more convenient to configure
```
Comment (WouterJ) [inline cookbook/email/gmail.rst]: it's the cookbook, so we can assume they are and we can say "It's more convenient to configure these options in the `parameters.yml.dist` file:"

Comment (WouterJ) [inline cookbook/email/gmail.rst]: actually, should be `parameters.yml`

**Comment by WouterJ** [comment]
Thank you @javiereguiluz! I've merged your PR and applied the final 2 comments in 6c07b60

**Pull request closed without merging**